cradle-cassandra/src/test/java/com/exactpro/cradle/cassandra/integration/messages/GroupedMessageIteratorProviderTest.java

```diff
@@ -0,0 +1,271 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.cradle.cassandra.integration.messages;
+
+import com.exactpro.cradle.*;
+import com.exactpro.cradle.cassandra.dao.messages.GroupedMessageIteratorProvider;
+import com.exactpro.cradle.cassandra.integration.BaseCradleCassandraTest;
+import com.exactpro.cradle.cassandra.integration.CassandraCradleHelper;
+import com.exactpro.cradle.cassandra.dao.CassandraDataMapper;
+import com.exactpro.cradle.cassandra.dao.CassandraDataMapperBuilder;
+import com.exactpro.cradle.cassandra.dao.CassandraOperators;
+import com.exactpro.cradle.cassandra.resultset.CassandraCradleResultSet;
+import com.exactpro.cradle.cassandra.retries.SelectQueryExecutor;
+import com.exactpro.cradle.filters.FilterForGreater;
+import com.exactpro.cradle.messages.*;
+import com.exactpro.cradle.utils.CradleStorageException;
+import org.assertj.core.api.Assertions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+public class GroupedMessageIteratorProviderTest extends BaseCradleCassandraTest {
+
+    private static final Logger logger = LoggerFactory.getLogger(GroupedMessageIteratorProviderTest.class);
+    private static final String GROUP_NAME = "test_group";
+    private static final String FIRST_SESSION_ALIAS = "test_session_alias_first";
+    private static final String SECOND_SESSION_ALIAS = "test_session_alias_second";
+
+    private final long storeActionRejectionThreshold = new CoreStorageSettings().calculateStoreActionRejectionThreshold();
+
+    private List<GroupedMessageBatchToStore> data;
+    private List<StoredGroupedMessageBatch> storedData;
+    private CassandraOperators operators;
+    private ExecutorService composingService = Executors.newSingleThreadExecutor();
+
+    @BeforeClass
+    public void startUp () throws IOException, InterruptedException, CradleStorageException {
+        super.startUp(true);
+
+        setUpOperators ();
+        generateData();
+    }
+
+    private void setUpOperators() throws IOException, InterruptedException {
+        CassandraDataMapper dataMapper = new CassandraDataMapperBuilder(session).build();
+        operators = new CassandraOperators(dataMapper, CassandraCradleHelper.getInstance().getStorageSettings());
```

Comment:
Shouldn't we use the Cradle here as well? Because if we store data with something else we might see that the tests are passed but actually cradle storage has some problems

Reply:
Those tests were written by Luka, and I'm not sure what were his intentions for such approach, I'll try to find out